### ISSUE TITLE
Improve polling using gocron

### ISSUE THREAD

**Issue by mostafa** [grafana/okta-logs-collector#3]
Currently we're using a for-loop with a sleep to poll logs from Okta. This works as expected, yet it can be improved, both in terms of flexibility and scheduling of the polling function by using [gocron](https://github.com/grafana/okta-logs-collector/blob/29cbf43ea3715fef4fd48d93b288e579bd50673f/client.go#L64-L92).

**Comment by KyriosGN0** [comment]
i would like to take on this issue, is there smth i should know before attempting to implement it ?

**Comment by mostafa** [comment]
@KyriosGN0 Good idea! However, I am not maintaining this anymore. Try my other project, [Helr](https://github.com/timescale/helr), if you like.

@jamesc-grafana Feel free to take it from here.
